From e5311fbfd5ad83671c61473d7acf4ddaf157e994 Mon Sep 17 00:00:00 2001
From: Jan Cholasta <jcholast@redhat.com>
Date: Thu, 23 Feb 2017 13:04:19 +0000
Subject: [PATCH] compat: fix `Any` params in `batch` and `dnsrecord`

The `methods` argument of `batch` and `dnsrecords` attribute of `dnsrecord`
were incorrectly defined as `Str` instead of `Any`.

https://fedorahosted.org/freeipa/ticket/6647

Reviewed-By: Martin Basti <mbasti@redhat.com>
---
 ipaclient/remote_plugins/2_114/batch.py | 2 +-
 ipaclient/remote_plugins/2_114/dns.py   | 2 +-
 ipaclient/remote_plugins/2_156/batch.py | 2 +-
 ipaclient/remote_plugins/2_156/dns.py   | 2 +-
 ipaclient/remote_plugins/2_164/batch.py | 2 +-
 ipaclient/remote_plugins/2_164/dns.py   | 2 +-
 ipaclient/remote_plugins/2_49/batch.py  | 2 +-
 ipaclient/remote_plugins/2_49/dns.py    | 2 +-
 8 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/ipaclient/remote_plugins/2_114/batch.py b/ipaclient/remote_plugins/2_114/batch.py
index 4a613b677bedda447a07d3d0bdc10d38762ccc61..2709e5907f18f254f7e605beff9a7f3c9a2ae18d 100644
--- a/ipaclient/remote_plugins/2_114/batch.py
+++ b/ipaclient/remote_plugins/2_114/batch.py
@@ -50,7 +50,7 @@ class batch(Command):
     NO_CLI = True
 
     takes_args = (
-        parameters.Str(
+        parameters.Any(
             'methods',
             required=False,
             multivalue=True,
diff --git a/ipaclient/remote_plugins/2_114/dns.py b/ipaclient/remote_plugins/2_114/dns.py
index 5d91dbcb37fcb42cb67ab76a1871fd3df6217cf8..acb8a658204fb18b088766f947f82839f053cbf3 100644
--- a/ipaclient/remote_plugins/2_114/dns.py
+++ b/ipaclient/remote_plugins/2_114/dns.py
@@ -326,7 +326,7 @@ class dnsrecord(Object):
             'dnsclass',
             required=False,
         ),
-        parameters.Str(
+        parameters.Any(
             'dnsrecords',
             required=False,
             label=_(u'Records'),
diff --git a/ipaclient/remote_plugins/2_156/batch.py b/ipaclient/remote_plugins/2_156/batch.py
index 4a613b677bedda447a07d3d0bdc10d38762ccc61..2709e5907f18f254f7e605beff9a7f3c9a2ae18d 100644
--- a/ipaclient/remote_plugins/2_156/batch.py
+++ b/ipaclient/remote_plugins/2_156/batch.py
@@ -50,7 +50,7 @@ class batch(Command):
     NO_CLI = True
 
     takes_args = (
-        parameters.Str(
+        parameters.Any(
             'methods',
             required=False,
             multivalue=True,
diff --git a/ipaclient/remote_plugins/2_156/dns.py b/ipaclient/remote_plugins/2_156/dns.py
index 39a0b269533481bcb5b193ad8a463a48146e5275..bbfaa9fd0fb2b582430a5c85761af206d53884f9 100644
--- a/ipaclient/remote_plugins/2_156/dns.py
+++ b/ipaclient/remote_plugins/2_156/dns.py
@@ -326,7 +326,7 @@ class dnsrecord(Object):
             'dnsclass',
             required=False,
         ),
-        parameters.Str(
+        parameters.Any(
             'dnsrecords',
             required=False,
             label=_(u'Records'),
diff --git a/ipaclient/remote_plugins/2_164/batch.py b/ipaclient/remote_plugins/2_164/batch.py
index 4a613b677bedda447a07d3d0bdc10d38762ccc61..2709e5907f18f254f7e605beff9a7f3c9a2ae18d 100644
--- a/ipaclient/remote_plugins/2_164/batch.py
+++ b/ipaclient/remote_plugins/2_164/batch.py
@@ -50,7 +50,7 @@ class batch(Command):
     NO_CLI = True
 
     takes_args = (
-        parameters.Str(
+        parameters.Any(
             'methods',
             required=False,
             multivalue=True,
diff --git a/ipaclient/remote_plugins/2_164/dns.py b/ipaclient/remote_plugins/2_164/dns.py
index b07a94f1942e3913d6d169b61d84a3b3db268671..244be87f32db6664e5264038b97bc53b704ff166 100644
--- a/ipaclient/remote_plugins/2_164/dns.py
+++ b/ipaclient/remote_plugins/2_164/dns.py
@@ -326,7 +326,7 @@ class dnsrecord(Object):
             'dnsclass',
             required=False,
         ),
-        parameters.Str(
+        parameters.Any(
             'dnsrecords',
             required=False,
             label=_(u'Records'),
diff --git a/ipaclient/remote_plugins/2_49/batch.py b/ipaclient/remote_plugins/2_49/batch.py
index a1f351d332d56c959bf8632cb218de8540f45005..67e5978e634b71735c1940086a80943d967ff1f6 100644
--- a/ipaclient/remote_plugins/2_49/batch.py
+++ b/ipaclient/remote_plugins/2_49/batch.py
@@ -50,7 +50,7 @@ class batch(Command):
     NO_CLI = True
 
     takes_args = (
-        parameters.Str(
+        parameters.Any(
             'methods',
             required=False,
             multivalue=True,
diff --git a/ipaclient/remote_plugins/2_49/dns.py b/ipaclient/remote_plugins/2_49/dns.py
index 07cef75c2a97c07a77a9ffa3997ec6fa431e3151..4b543a2c2539f7b67467b0a38ab8013a1ebe0840 100644
--- a/ipaclient/remote_plugins/2_49/dns.py
+++ b/ipaclient/remote_plugins/2_49/dns.py
@@ -256,7 +256,7 @@ class dnsrecord(Object):
             label=_(u'Class'),
             doc=_(u'DNS class'),
         ),
-        parameters.Str(
+        parameters.Any(
             'dnsrecords',
             required=False,
             label=_(u'Records'),
-- 
2.9.3

